src/Data/Loader/UserLoader.php

```diff
@@ -72,15 +72,13 @@ public function get_public_users( array $keys ) {
 		$public_only = true;
 
 		$where = get_posts_by_author_sql( $post_types, true, $author_id, $public_only );
-		$ids   = implode( ', ', array_fill( 0, count( $keys ), '%d' ) );
-		$count = count( $keys );
+		$ids   = implode( ', ', $keys );
 
 		global $wpdb;
 
-		$results = $wpdb->get_results( // phpcs:ignore WordPress.DB.DirectDatabaseQuery.NoCaching,WordPress.DB.DirectDatabaseQuery.DirectQuery
+		$results = $wpdb->get_results( // phpcs:ignore WordPress.DB.DirectDatabaseQuery.DirectQuery,WordPress.DB.DirectDatabaseQuery.NoCaching
 			$wpdb->prepare(
-				"SELECT DISTINCT `post_author` FROM $wpdb->posts $where AND `post_author` IN ( $ids ) LIMIT $count", // phpcs:ignore WordPress.DB.PreparedSQL.InterpolatedNotPrepared,WordPress.DB.PreparedSQLPlaceholders.UnfinishedPrepare
-				$keys
+				"SELECT DISTINCT `post_author` FROM $wpdb->posts INNER JOIN $wpdb->users ON $wpdb->posts.post_author = $wpdb->users.ID $where AND `post_author` IN ( $ids )" // phpcs:ignore WordPress.DB.PreparedSQL.InterpolatedNotPrepared,WordPressVIPMinimum.Variables.RestrictedVariables.user_meta__wpdb__users
```

Comment:
I don't know why, but selecting `wp_users.ID` returns the query results much faster. It's not a big difference but I just wanted to share.

```
time mysql -e "SELECT DISTINCT wp_2_posts.post_author FROM wp_2_posts INNER JOIN wp_users ON wp_2_posts.post_author = wp_users.ID WHERE ( ( post_type = 'post' AND(post_status = 'publish') ) OR ( post_type = 'page' AND(post_status = 'publish') ) OR( post_type = 'place' AND(post_status = 'publish') ) ) AND post_author IN (1, 2, 3, 4, 5);"
+-------------+
| post_author |
+-------------+
|           1 |
|          2 |
|         3 |
|         4 |
+-------------+

real	0m0.471s
user	0m0.008s
sys	0m0.000s
```

```
time mysql -e "SELECT DISTINCT wp_users.ID FROM wp_2_posts INNER JOIN wp_users ON wp_2_posts.post_author = wp_users.ID WHERE ( ( post_type = 'post' AND(post_status = 'publish') ) OR ( post_type = 'page' AND(post_status = 'publish') ) OR( post_type = 'place' AND(post_status = 'publish') ) ) AND post_author IN (1, 2, 3, 4, 5);"
+-----+
| ID  |
+-----+
|   1 |
|  2 |
| 3 |
| 4 |
+-----+

real	0m0.010s
user	0m0.004s
sys	0m0.004s
```

Reply:
@sathyapulse Very nice! I've made adjustments based on this suggestion and pushed up the changes. Thanks for digging into this and providing constructive feedback to make this better and faster for everyone! 🙏🏻 🙌🏻